src/App.tsx

```diff
@@ -15,48 +16,131 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const getReorderedGoods = useCallback(
+    (
+      goods: string[],
+      {
+        sortType: reorderSortType,
+        isReversed: reorderIsReversed,
+      }: ReorderOptions,
+    ) => {
```

Comment:
this part is hard to read, you can improve it a little bit
```suggestion
  const getReorderedGoods = useCallback((
      goods: string[],
      {
        sortType: reorderSortType,
        isReversed: reorderIsReversed,
      }: ReorderOptions,
    ) => {
```